### PR TITLE
Web improvements Merge

### PR DESCRIPTION
**Pull request by codingsamuel**
### Possible to merge Web-improvements to master for 1.2
Added devextreme with graphs for polls
Added dark and light theme optimization
Loader animation
Fixed Stream announcer saving

### CODE REVIEW

File: src/Volvox.Helios.Web/Views/Settings/StreamerSettings.cshtml [hunk 1]
```diff
@@ -9,49 +9,150 @@
...
-                <input asp-for="ChannelSettings.Enabled" class="form-check-input" />
-                <label asp-for="ChannelSettings.Enabled" class="form-check-label"></label>
-                <span asp-validation-for="ChannelSettings.Enabled" class="text-danger"></span>
-            </div>
-            <div class="form-check form-group">
-                <input asp-for="ChannelSettings.RemoveMessages" class="form-check-input" />
-                <label asp-for="ChannelSettings.RemoveMessages" class="form-check-label"></label>
-                <span asp-validation-for="ChannelSettings.RemoveMessages" class="text-danger"></span>
-            </div>
+                    @*<div class="col-12">
```
[the remark this answers is not in the document]
Comment (BillChirico) [reply src/Volvox.Helios.Web/Views/Settings/StreamerSettings.cshtml]: Good point

File: src/Volvox.Helios.Web/wwwroot/css/loader.min.css [hunk 2]
```diff
@@ -0,0 +1 @@
+﻿#preloader{display:flex;position:fixed;background:#000;color:#fff;width:100%;height:100%;top:0;left:0;z-index:10000;justify-content:center;align-items:center}#preloader .loader-anim{position:relative;padding:10px 30px;border:3px solid #5DCC32;border-radius:3px;overflow:hidden}#preloader.toggled{transition:all 1s;opacity:0}#preloader .loader-anim .bg{animation:slide 1s ease-in-out infinite forwards;background:#5DCC32;height:100%;width:100%;position:absolute;left:-100%;top:0;z-index:-1;}@keyframes slide{0%{left:-100%}100%{left:101%}}
```
Comment (HurricanKai) [inline src/Volvox.Helios.Web/wwwroot/css/loader.min.css]: All .min files do not need to be in the git, they are generated on build by the bundler anyways.

File: src/Volvox.Helios.Web/wwwroot/css/site.min.css [hunk 3]
```diff
@@ -1 +1 @@
...
...ame_cont.HW_visible{z-index:100000}#changelog-wrapper{position:absolute;width:100%;height:100%;top:0;left:0;z-index:1}#changelog #HW_badge_cont{position:absolute;width:100%;height:100%}#changelog #HW_badge{left:17px}.poll-item{border-color:var(--color-volvox)}.vote-count{position:absolute;right:20px;background:var(--color-volvox);top:0;color:#fff;justify-content:center;padding:0 14px 2px;border-radius:0 0 3px 3px}.footer-icon{width:40px;height:40px;justify-content:center;border:2px solid var(--color-volvox);color:#fff;transition:.5s}.footer-icon:hover{background:none;color:var(--color-volvox)}.dx-editor-outlined{border-color:#ced4da;height:38px;transition:border-color .15s ease-in-out,box-shadow .15s ease-in-out}.dx-texteditor.dx-state-hover,.dx-texteditor.dx-state-focused{border-color:var(--color-volvox) !important}.dx-list-item.dx-state-active{background:var(--color-volvox) !important}#cookieConsent{border-top:3px solid var(--color-skyblue)}@media screen and (max-width:767px){#themeToggler{display:none}}@media screen and (max-width:575px){.material-badge{display:block;margin-bottom:10px}.home-header{min-height:50vh}#HW_frame_cont{width:calc(100% - 20px) !important;margin:0 10px}}
```
Comment (HurricanKai) [inline src/Volvox.Helios.Web/wwwroot/css/site.min.css]: All .min files do not need to be in the git, they are generated on build by the bundler anyways.

File: src/Volvox.Helios.Web/Views/Remembot/EditRecurringReminder.cshtml [hunk 4]
```diff
@@ -22,85 +22,80 @@
...
-                    <input type="button" id="deleteBtn" value="Delete Reminder" class="btn btn-danger" />
-                }
-            </div>
-        </form>
+    <div asp-validation-summary="ModelOnly" class="text-danger"></div>
+        
+    <div class="row mt-4">
+        <div class="ml-auto">
+            <button type="submit" class="btn btn-volvox-outline pr-4 pl-4" style="border-radius:20px;">Save Reminder</button>
+            @if (Model.Id != default(long))
```
[the remark this answers is not in the document]
Comment (mhallmark) [reply src/Volvox.Helios.Web/Views/Remembot/EditRecurringReminder.cshtml]: Maybe the view name is a misnomer. This "edit" page is used for both adding and editing an existing reminder.

File: src/Volvox.Helios.Web/Views/Shared/_Layout.cshtml [hunk 5]
```diff
@@ -57,11 +95,26 @@
...
+                    <a href="https://github.com/VolvoxCommunity/Volvox.Helios/" class="bg-volvox d-inline-flex footer-icon valign rounded-circle mr-2" target="_blank">
+                        <i class="fab fa-github"></i>
+                    </a>
+                    <a href="https://discord.gg/jReSmc3" class="bg-volvox d-inline-flex valign rounded-circle footer-icon" target="_blank">
+                        <i class="fab fa-discord"></i>
+                    </a>
+                </div>
+                <hr class="border-volvox col-md-6 mt-4">
+                <p class="mb-0 text-center">
+                    &copy; 2018 - Volvox.Helios v@(Assembly.GetEntryAssembly().GetName().Version.Major).@(Assembly.GetEntryAssembly().GetName().Version.Minor).@(Assembly.GetEntryAssembly().GetName().Version.Build)
```
Comment (HurricanKai) [inline src/Volvox.Helios.Web/Views/Shared/_Layout.cshtml]: Im quite shure this will result in multiple Assembly.GetEntryAssembly().GetName() calls?

Comment (BillChirico) [reply src/Volvox.Helios.Web/Views/Shared/_Layout.cshtml]: Will add to a variable

File: src/Volvox.Helios.Web/Views/Remembot/EditRecurringReminder.cshtml (hunk 4 above)
[the remark this answers is not in the document]
Comment (HurricanKai) [reply src/Volvox.Helios.Web/Views/Remembot/EditRecurringReminder.cshtml]: Ah, i see 👍

File: src/Volvox.Helios.Core/Modules/RemembotModule/RemembotModule.cs [hunk 6]
```diff
@@ -53,6 +53,18 @@ public class RemembotModule : Module
...
+        /// <summary>
+        ///     Returns true if the module is enabled for the specified guild and false if not.
+        /// </summary>
+        /// <param name="guildId">Id if the guild to check.</param>
+        /// <returns>True if the module is enabled for the specified guild and false if not.</returns>
+        public override async Task<bool> IsEnabledForGuild(ulong guildId)
+        {
+            var settings = await _moduleSettings.GetSettingsByGuild(guildId);
+
+            return settings != null && settings.Enabled;
```
Comment (HurricanKai) [inline src/Volvox.Helios.Core/Modules/RemembotModule/RemembotModule.cs]: I personally prefer the short syntax, `settings?.Enabled ?? false` but personal preference

File: src/Volvox.Helios.Core/Modules/Command/CommandManager.cs [hunk 7]
```diff
@@ -33,6 +33,16 @@ public class CommandManager : Module
 
         private IList<ICommand> Modules { get; }
 
+        /// <summary>
+        ///     Returns true if the module is enabled for the specified guild and false if not.
+        /// </summary>
+        /// <param name="guildId">Id if the guild to check.</param>
+        /// <returns>True if the module is enabled for the specified guild and false if not.</returns>
+        public override Task<bool> IsEnabledForGuild(ulong guildId)
```
Comment (HurricanKai) [inline src/Volvox.Helios.Core/Modules/Command/CommandManager.cs]: Possibly ValueTask<bool> will be of use here?

**Pull request merged**